### ISSUE TITLE
Rscript FitHiC_SigInt.r exits prematurely with try-catch error

### ISSUE THREAD

**Issue by jaavedm** [ay-lab/FitHiChIP#49]
I'm having problems getting FitHiChip to finish. It gets to a point in the **FitHiChIP_HiCPro.sh** script where it starts calling the Rscript **FitHiC_SigInt.r**. However, in this Rscript, there seems to be a bug in the code that does not handle my input files correctly. The Rscript eventually dies with the following error:

```
Error in as.data.frame.default(x[[i]], optional = TRUE, stringsAsFactors = stringsAsFactors) :
  cannot coerce class ‘"try-error"’ to a data.frame
Calls: as.data.frame ... <Anonymous> -> as.data.frame -> as.data.frame.default
In addition: Warning message:
In parallel:::mclapply(1:length(avg_int_dist), mc.cores = ncore,  :
  all scheduled cores encountered errors in user code
Execution halted
```

Because of this error FitHichip ends prematurely saying that no significant interactions are found. 

I don't have the skills to debug where the error is occurring in **FitHiC_SigInt.r** but I have made a standalone input files that allow you to reproduce the error and debug the script. 

Here is my command line:
```
Rscript /opt/apps/FitHiChIP/src/FitHiC_SigInt.r \
--InpFile Interactions.sortedGenDist.bed \
--headerInp \
--OutFile fithichip.interactions_FitHiC.bed \
--CoverageFile fithichip.coverage_ICE_Bias.bed \
--BinSize 20000 \
--P2P 0 \
--IntType 3 \
--UseNonzeroContacts 0 \
--BiasCorr 1 \
--BiasType 2 \
--BiasLowThr 0.2 \
--BiasHighThr 5 \
--Draw \
--cccol 7 \
--BiasFilt 0 \
--MultBias 0 \
--Resid 0 \
--EqOcc 1
```

I have attached a gzip'ed copy of the input files "**Interactions.sortedGenDist.bed**" and "**fithichip.coverage_ICE_Bias.bed**"

**Comment by jaavedm** [comment]
Github is preventing me from uploading large files. Here are the input files:
http://45.132.241.75/public/Interactions.sortedGenDist.bed.gz
http://45.132.241.75/public/fithichip.coverage_ICE_Bias.bed.gz

**Comment by jaavedm** [comment]
I think I found the error(s). For reasons that I've not determined yet, some of the vectors in this R script are of character class rather than numeric. Therefore, some of the log10 operations in the script fail. 

For example on Line 988, I made the following changes:

`bias_regression_data <- cbind.data.frame(log10(as.numeric(Curr_Int_Set_BiasThr[, bias1.col])), log10(as.numeric(Curr_Int_Set_BiasThr[, bias2.col])), (log10(as.numeric(Curr_Int_Set_BiasThr[, opt$cccol])) - logdistCCVec))		`

Basically I'm forcing the vectors to be numeric. I'm not sure if this is masking some bigger problem, so please let me know once you carefully evaluate my input files that reproduces the error. 

With these `log10(as.numeric()) `changes throughout the script, the script runs to completion.

**Comment by ay-lab** [comment]
Hi. Glad you found the error. For many reasons, we would still suggest that you use Python implementation of FitHiC2. R implementation is very outdated and may have bugs. Any questions about the python version, we can help

**Comment by jaavedm** [comment]
Hello again. Is it now recommended to process hichip data with fithic2 rather than FitHiChIP. I'm seeing some conflicting messages about this and want to reach out to you about this. 

For example, I see this [issue ](https://github.com/ay-lab/fithic/issues/23) on the fithic github page from Apr 2019 recommending using FitHiChIP for Hichip data.

**Comment by ay-lab** [comment]
Hi Javeed. Yes, it is better if you use FitHiChIP for HiChIP data.

**Comment by ay-lab** [comment]
FitHiC2 is pretty much the same statistical model as FitHiC but just a better implementation and a few new features added. FitHiChIP is specifically designed with the features of HiChIP data in mind. 
In terms of our tools I suggest:
FitHiC2 or Mustache for Hi-C data. 
FitHiChIP for HiChIP data. It is applicable to other enrichment-based assays such as promoter-capture Hi-C or ChIA-PET as well.

**Comment by jaavedm** [comment]
Thank you. I will continue to use FitHiChIP then for my hichip analysis. I appreciate the clarification.
